#!/usr/bin/env python
# PathFinder.cfg
PACKAGE = "pmadmu_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("static_time_tolerance", double_t, 0, "Estimation of how long the robot needs to travel between two grid spaces. Will add a static length to the 'snakes'", 4.0, 0.0, 50.0)
gen.add("dynamic_time_tolerance", double_t, 0, "Estimation of motion uncertainty. Lets 'snakes' grow over time.", 1.5, 0.0, 10.0)
gen.add("speed", double_t, 0, "Robot speed.", 0.5, 0.0, 5.0)

gen.add("allow_straights", bool_t, 0, "Allows the following movements: 0°, 90°, 180°, 360°.", True)
gen.add("allow_diagonals", bool_t, 0, "Allows the following movements: 45°, 135°, 225°, 315°.", True)
gen.add("allow_knight_moves", bool_t, 0, "Allows the following movements: 26°, 63°, 116°, 153°, 206°, 243°, 296°, 333° (like a knight in chess).", False)

gen.add("check_dynamic_obstacles", bool_t, 0, "Check for dynamic obstacles.", True)
gen.add("dynamic_visualization", bool_t, 0, "Publishes timing map after every step, very expensive.", False)


exit(gen.generate(PACKAGE, "central_controller", "PathFinder"))